core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ActiveDirectoryUserStoreManager.java

```diff
@@ -625,6 +628,142 @@ protected void doSetUserAttribute(String userName, String attributeName, String
         }
     }
 
+    @Override
+    public String[] getUserListOfLDAPRole(RoleContext context, String filter) throws UserStoreException {
+
+        boolean debug = logger.isDebugEnabled();
+
+        if (debug) {
+            logger.debug("Getting user list of role: " + context.getRoleName() + " with filter: " + filter);
```

Comment:
If we are to append more than one variable to a String, better to use `String.format`. WDYT?

Reply:
Shall we fix this in other places as well?

Reply:
Not necessary as String.format does not have a performance benefit over concatenation. Refer the second answer in [1].
Also single line concatenations are optimised [2] and converted to StringBuilder under the hood. Memory wise is the same thing and to stick with the same pattern in the class I am avoiding the use of StringBuilder.

[1] https://stackoverflow.com/questions/925423/is-it-better-practice-to-use-string-format-over-string-concatenation-in-java
[2] https://stackoverflow.com/questions/18453458/string-builder-vs-string-concatenation